### ISSUE TITLE
ACCESSIBILITY : Anchor Tags are not accessible on Android & IOS When screen reader is present.

### ISSUE THREAD

**Issue by FrancisLeigh** [meliorence/react-native-render-html#285]
When screen readers are active on my devices (IOS & Android) I cannot seem to select the anchor-tag links that are visible on the page. 

Has anyone had similar issues with this? I feel Accessibility is a bit of a dark-art when it comes to React-Native.

```
"react-native": "~0.59.0",
"react-native-render-html": "^4.1.2",
```

`IOS (Physical) & Android (Emulator)`

**Comment by linelson** [comment]
Also experiencing this bug, but on RN 62.2.

**Comment by jsamr** [comment]
@FrancisLeigh @linelson Accessibility is a very serious matter and I believe we should have first-class support in the Foundry redesign (v6). I am not very knowledgeable in the field, and I would love input on how we can improve things more generally. The best would be that someone takes the time to think about and offers a RFC that we would implement in Foundry.

**I'm calling the community to participate**, we provide [guidelines here](https://github.com/archriss/react-native-render-html/blob/master/CONTRIBUTING.adoc#request-for-comments), and you can take inspiration [from this one](https://github.com/archriss/react-native-render-html/blob/master/rfc/001-A-deterministic-approach-to-images-scaling.adoc).

**Comment by FrancisLeigh** [comment]
A part-time solution that we came up with of was to show an "appendix of links" below any content block that renders an anchor when the screen reader is enabled. 

We achieved this by parsing the incoming HTML for anchor tags, taking the tags content and corresponding links and rendering within a list below the content block.

Obviously not a solid solution but if anyone finds themselves here and needs a UX to handle the internal links...

**Comment by jsamr** [comment]
This issue should be fixed in the Foundry pre-release (alpha.15), but please let me know if there are still problems remaining. See #430 for instructions on how to try it out.

**Comment by stevehanson** [comment]
I was not able to find any workaround for this issue that works across screen readers, voice control, and keyboard users without adversely affecting the experience of users who are not using assistive tech. Ultimately, after trying to develop a workaround for several days, I switched my app to use a webview instead of this library, and I now have a fully accessible app. This is an amazing library, and I have nothing but gratitude for the contributors who have built it. It was a shame to have to move away from it due to this React Native issue, but I'd recommend the same for anyone building an app with nested links, because they will not be accessible.

I've posted more information about my experiences with this in the RN Render HTML Discord for anyone facing similar issues. [Link to Discord thread](https://discord.com/channels/736906960041148476/736906960041148480/1168637276478246932) (have to first join the Discord -- see instructions in Readme).

**Comment by YashDOM** [comment]
I have tried to use custom renderer to solve the accessibility issue, and I have almost solve it. But the problem now is I am unable to solve the inline alignment issue. If someone can help it would be great. Here is my custom rendere which I used : 

`const LinkRenderer: React.FC<CustomRendererProps<TNode>> = props => {
      const {tnode, TDefaultRenderer} = props;
      const href = tnode.attributes?.href ?? '#';
      let linkText = 'Link';
      const firstChild = tnode.children[0];

      // Check if the first child is a TText node
      if (firstChild?.type === 'text') {
        const textNode = firstChild as TText;
        linkText = textNode.data;
      }

      const handlePress = () => {
        if (href !== '#') {
          Linking.openURL(href).catch(err => {
            console.warn('Failed to open link:', err);
          });
        }
      };

      return (
        <Pressable
          onPress={() => {
            if (href) Linking.openURL(href);
          }}
          accessible={true}
          accessibilityRole="link"
          focusable={true}
          style={{
            display: 'flex',
            flexDirection: 'row',
            alignItems: 'baseline', // align with text
            padding: 0,
            margin: 0,
            borderWidth: 0,
            // transform: [{translateY: Platform.OS === 'ios' ? 2 : 5}],
          }}>
          <TDefaultRenderer {...props} />
        </Pressable>
      );
    };`

I am able to handle the inline alignment with  transform: [{translateY: Platform.OS === 'ios' ? 2 : 5}], but it is buggy if I change the orientation.
